### PR TITLE
fixes to use the odbc_bridge_use_connection_pooling setting correctly

### PR DESCRIPTION
**Pull request by bharatnc**
This is a follow up to #37705. The prior fix wasn't actually working and this PR adds follow up fixes and a test for enabling/disabling the odbc connection pooling.

### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fixes to correctly use `odbc_bridge_use_connection_pooling` setting.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: programs/odbc-bridge/SchemaAllowedHandler.cpp [hunk 1]
```diff
@@ -70,13 +70,19 @@ void SchemaAllowedHandler::handleRequest(HTTPServerRequest & request, HTTPServer
         return;
     }
 
+     bool use_connection_pooling = params.getParsed<bool>("use_connection_pooling", true);
```
Comment (kssenii) [inline programs/odbc-bridge/SchemaAllowedHandler.cpp:73]: isn't it more correct to name it "use_connection_pool"?

File: src/BridgeHelper/XDBCBridgeHelper.h [hunk 2]
```diff
@@ -63,10 +63,12 @@ class XDBCBridgeHelper : public IXDBCBridgeHelper
     XDBCBridgeHelper(
             ContextPtr context_,
             Poco::Timespan http_timeout_,
-            const std::string & connection_string_)
+            const std::string & connection_string_,
+            bool use_connection_pooling_=true)
```
Comment (kssenii) [inline src/BridgeHelper/XDBCBridgeHelper.h:67]: let's get rid of default value, to set it explicitly everywhere with setting.

Comment (bharatnc) [reply src/BridgeHelper/XDBCBridgeHelper.h:67]: sure I will fix it.

File: programs/odbc-bridge/SchemaAllowedHandler.cpp (hunk 1 above)
Comment (bharatnc) [reply programs/odbc-bridge/SchemaAllowedHandler.cpp:73]: I don't have any strong opinions about this - to me they are more are less similar in meaning.

Comment (kssenii) [reply programs/odbc-bridge/SchemaAllowedHandler.cpp:73]: Ok

File: src/BridgeHelper/XDBCBridgeHelper.h (hunk 2 above)
Comment (kssenii) [reply src/BridgeHelper/XDBCBridgeHelper.h:67]: @bharatnc,  let's do? Waiting for this

Comment (bharatnc) [reply src/BridgeHelper/XDBCBridgeHelper.h:67]: sure, working on it, I'll update the PR soon.

Comment (bharatnc) [reply src/BridgeHelper/XDBCBridgeHelper.h]: fixed.

**Review by kssenii — APPROVED** [review]

**Pull request merged**